community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/checkpoint/CheckPointScheduler.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.transaction.log.checkpoint;
+
+import java.io.IOException;
+
+import org.neo4j.kernel.impl.store.UnderlyingStorageException;
+import org.neo4j.kernel.impl.util.JobScheduler;
+import org.neo4j.kernel.lifecycle.LifecycleAdapter;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.neo4j.kernel.impl.util.JobScheduler.Groups.checkPoint;
+
+public class CheckPointScheduler extends LifecycleAdapter
+{
+    private final CheckPointer checkPointer;
+    private final JobScheduler scheduler;
+    private final long recurringPeriodMillis;
+    private final Runnable job = new Runnable()
+    {
+        @Override
+        public void run()
+        {
+            try
+            {
+                checkPointer.checkPointIfNeeded();
+            }
+            catch ( IOException e )
+            {
+                throw new UnderlyingStorageException( e );
```

Comment:
I will add a `kernel.panic()` in `CheckPointerImpl` in case it fails to write the check point in the log like: https://github.com/neo4j/neo4j/blob/2.3/community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/BatchingTransactionAppender.java#L241. Don't see how rescheduling will help here.